bin/pycbc_create_injections

```diff
@@ -189,23 +200,56 @@ dists = distributions.read_distributions_from_config(cp, opts.dist_section)
 randomsampler = JointDistribution(variable_params, *dists,
                                **{"constraints" : constraints})
 
-logging.info("Drawing samples")
-samples = randomsampler.rvs(size=opts.ninjections)
-
-if waveform_transforms is not None:
-    logging.info("Transforming to waveform transform parameters")
-    for t in waveform_transforms:
-        if not set(t.inputs).isdisjoint(set(static_params.keys())):
-            for item in list((set(t.inputs) & set(static_params.keys())
-                             - set(samples.fieldnames))):
-                samples = samples.add_fields([numpy.repeat(static_params[item],
-                                             opts.ninjections).astype(float)],
-                                             [item])
-    samples = transforms.apply_transforms(samples, waveform_transforms)
-    write_args = [arg for arg in samples.fieldnames
-                  if arg not in static_params.keys()]
+if opts.ninjections:
+    draw_size = opts.ninjections
+else:
+    draw_size = 4000 # Just a default so it's not super slow drawing large sets
+    old_samples = None
+
+while True:
+    logging.info("Drawing samples")
+    samples = randomsampler.rvs(size=draw_size)
+
+    if waveform_transforms is not None:
+        logging.info("Transforming to waveform transform parameters")
+        for t in waveform_transforms:
+            if not set(t.inputs).isdisjoint(set(static_params.keys())):
+                for item in list((set(t.inputs) & set(static_params.keys())
+                                 - set(samples.fieldnames))):
+                    samples = samples.add_fields([numpy.repeat(static_params[item],
+                                                 draw_size).astype(float)],
+                                                 [item])
+        samples = transforms.apply_transforms(samples, waveform_transforms)
+
+    # We are drawing until we hit a time so check if we've reached it
+    if opts.gps_start_time:
+        if 'tc' in samples.fieldnames:
+            raise RuntimeError("A distribution was given for 'tc'. This is "
+                               "not compatible with providing start/end times.")
+
+        if old_samples is not None:
+            samples = numpy.concatenate([old_samples, samples])
+
+        old_samples = samples
+
+        # We have enough to cover out time span so we can fixup the times
+        # and stop here
+        if samples['time_delay'].sum() > opts.gps_end_time - opts.gps_start_time:
```

Comment:
This will fail with a non-obvious error if `time_delay` isn't included in the config file.

What's the reason for wanting this to be in the config file? Are you thinking of using a different distribution then uniform? I still find it a bit weird that you have to have half the functionality on the command line and half in the config file, but if you do think a non-uniform distribution will ever be useful, then fine. However, call it `delta_tc`; I think that's a bit more descriptive. And add a check that it's provided (just like you checked that `tc` is not provided).